.gitignore

```diff
@@ -10,7 +10,6 @@ compare-results/*
 *.user
 *.sln.docstates
 .vs
-developer/
```

Comment:
yup, so, it looks like this makes the "developer/" folder not be tracked, and since these doc files are in there, I remove it. I don't think we want to ignore these files, do we? Maybe this was leftover from previous autorest structure? feel free to check with some other autorest folks on this as needed, as I don't know why we could want to ignore things under "developer".